aws_ecr_scan_results.py

```diff
@@ -3,21 +3,24 @@
 import argparse
 import json
 import os
+import inspect
 import requests
 import boto3
 
 
 class ECRScanChecker:
     """Check ECR Scan results for all service container images."""
     aws_account_id = ''
+    iam_role_name = ''
     aws_iam_session = ''
     aws_ecr_client = ''
     images_to_check = []
     tag = ''
     report = ''
     report_limit = ''
 
-    def __init__(self, report_limit, search_term):
+    def __init__(self, iam_role_name, report_limit, search_term):
+        self.iam_role_name = iam_role_name
         self.report_limit = int(report_limit)
         self.aws_account_id = 311462405659  # management account id
```

Comment:
Probably worth extracting this too, so its reusable not by us.

Reply:
I'd like to do that in another change in the future. Beyond making this ok for opg projects to use, there's much more that needs to be made configurable, so I'm working out how best to provide those changes which will be a fair bit of refactoring.

Reply:
Nice!